src/pwncore/routes/auth.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import jwt
+from fastapi import APIRouter, Header, Response, HTTPException
+from pwncore.models import Team
+from pwncore.config import config
+import datetime
+from passlib.hash import bcrypt
+from pydantic import BaseModel
+from tortoise.transactions import atomic
+
+# Metadata at the top for instant accessibility
+metadata = {
+    "name": "auth",
+    "description": "Authentication using a JWT using a single access token.",
+}
+
+router = APIRouter(prefix="/auth", tags=["auth"])
+
+
+class SignupBody(BaseModel):
+    name: str
+    password: str
+    # members: list[str]
+
+
+class LoginBody(BaseModel):
+    name: str
+    password: str
+
+
+@atomic()
+@router.post("/signup")
+async def signup_team(team: SignupBody, response: Response):
+    try:
+        if await Team.exists(name=team.name):
+            response.status_code = 406
+            return {"msg_code": config.msg_codes["team_exists"]}
+
+        # TODO: Add users details
+
+        await Team.create(name=team.name, secret_hash=bcrypt.hash(team.password))
+    except Exception:
+        response.status_code = 500
+        return {"msg_code": config.msg_codes["db_error"]}
+    return {"msg_code": config.msg_codes["signup_success"]}
+
+
+@router.post("/login")
+async def team_login(team_data: LoginBody, response: Response):
+    # TODO: Simplified logic since we're not doing refresh tokens.
+
+    team = await Team.get_or_none(name=team_data.name)
+    if team is None:
+        response.status_code = 404
+        return {"msg_code": config.msg_codes["team_not_found"]}
+    if not bcrypt.verify(team_data.password, team.secret_hash):
+        response.status_code = 401
+        return {"msg_code": config.msg_codes["wrong_password"]}
+
+    current_time = datetime.datetime.utcnow()
+    expiration_time = current_time + datetime.timedelta(hours=config.jwt_valid_duration)
+    token_payload = {"team_id": team.id, "exp": expiration_time}
+    token = jwt.encode(token_payload, config.jwt_secret, algorithm="HS256")
+
+    # Returning token to be sent as an authorization header "Bearer <TOKEN>"
+    return {
+        "msg_code": config.msg_codes["login_success"],
+        "access_token": token,
+        "token_type": "bearer",
+    }
+
+
+# Custom JWT processing (since FastAPI's implentation deals with refresh tokens)
+async def get_jwt(*, authorization: str = Header()):
+    token = authorization.split(" ")[1]  # Remove Bearer
```

Comment:
Unhandled IndexError

also prefer doing `authorization[<constant you find>:]` over splitting

Reply:
I think we should prefer splitting, since the expected authorization header should be in format `Bearer <token>`.
In case the user (frontend) passes the authorization header as just `<token>`, it would raise an error (I forgot to catch the error).

If we use `authorization[7:]` to strip the `Bearer `, it would not lead to an error in the case that Bearer does not exist in the header.

Reply:
If `Bearer ` does not exist in the token it would lead to an error by splitting too.
.removeprefix would do the required validation and deletion

Also, I just noticed but this function shouldn't be async, it is doing no IO